security/s2a/internal/crypter/aesgcm_test.go

```diff
@@ -132,114 +135,79 @@ func TestAESGCMKeySizeUpdate(t *testing.T) {
 	}
 }
 
-// Test Encrypt/Decrypt using an invalid nonce size.
-func TestAESGCMEncryptDecryptInvalidNonce(t *testing.T) {
-	key := make([]byte, aes128GcmKeySize)
-	crypter, err := NewAESGCM(key)
-	if err != nil {
-		t.Fatalf("NewAESGCM(keySize=%v) failed, err: %v", aes128GcmKeySize, err)
-	}
-	// Construct nonce with invalid size.
-	nonce := make([]byte, 1)
-	if _, err = crypter.Encrypt(nil, nil, nonce, nil); err == nil {
-		t.Errorf("Encrypt should fail due to invalid nonce size")
-	}
-	if _, err = crypter.Decrypt(nil, nil, nonce, nil); err == nil {
-		t.Fatalf("Decrypt should fail due to invalid nonce size")
-	}
-}
-
 // Test encrypt and decrypt on roundtrip messages for AES-GCM with and without
 // updating the keys.
 func TestAESGCMEncryptRoundtrip(t *testing.T) {
-	for _, shouldUpdateKeys := range []bool{true, false} {
-		for _, keySize := range []int{aes128GcmKeySize, aes256GcmKeySize} {
-			key := make([]byte, keySize)
-			sender, receiver := getGCMCryptoPair(key, t)
-			if shouldUpdateKeys {
-				// Update the key with a new one which is different from the original.
-				newKey := make([]byte, keySize)
-				newKey[0] = '\xbd'
-				if err := sender.UpdateKey(newKey); err != nil {
-					t.Fatalf("sender UpdateKey failed with: %v", err)
-				}
-				if err := receiver.UpdateKey(newKey); err != nil {
-					t.Fatalf("receiver UpdateKey failed with: %v", err)
-				}
-			}
-			testGCMEncryptRoundtrip(sender, receiver, t)
+	for _, keySize := range []int{aes128GcmKeySize, aes256GcmKeySize} {
```

Comment:
Sorry for not clarifying my comment. Let's keep the `for _, shouldUpdateKeys := range []bool{true, false}` loop.

Reply:
I don't think the `shouldUpdateKeys` loop will be useful here since we don't change anything between each roundtrip call. Testing back-to-back roundtrips in the S2A Half Connection will be useful since the nonce will be updated with each roundtrip, but it wouldn't be useful here since we don't update the nonce.